consumers/aws.go

```diff
@@ -108,26 +143,50 @@ func (a *awsClient) Sync(endpoints []*pkg.Endpoint) error {
 	if len(upsert) > 0 || len(del) > 0 {
 		log.Debugln("Records to be upserted: ", upsert)
 		log.Debugln("Records to be deleted: ", del)
-		return a.client.ChangeRecordSets(upsert, del, nil)
+		return a.client.ChangeRecordSets(upsert, del, nil, zoneID)
 	}
 
 	log.Infoln("No changes submitted")
-
 	return nil
 }
 
 func (a *awsClient) Process(endpoint *pkg.Endpoint) error {
+	hostedZonesMap, err := a.client.GetHostedZones()
+	if err != nil {
+		return err
+	}
+
 	aliasRecords, err := a.client.EndpointsToAlias([]*pkg.Endpoint{endpoint})
 	if err != nil {
 		return err
 	}
 
 	create := []*route53.ResourceRecordSet{aliasRecords[0], a.client.GetAssignedTXTRecordObject(aliasRecords[0])}
 
-	err = a.client.ChangeRecordSets(nil, nil, create)
-	if err != nil && strings.Contains(err.Error(), "it already exists") {
+	zoneID := getZoneIDForEndpoint(hostedZonesMap, aliasRecords[0])
+	if zoneID == "" {
+		log.Errorf("Hosted zone for endpoint: %s is not found. Skipping record...", endpoint.DNSName)
+		return nil
+	}
+
+	err = a.client.ChangeRecordSets(nil, nil, create, zoneID)
+	if err != nil && strings.Contains(err.Error(), "already exists") {
 		log.Warnf("Record [name=%s] could not be created, another record with same name already exists", endpoint.DNSName)
 		return nil
 	}
+
 	return err
 }
+
+//getZoneIDForEndpoint returns the zone id for the record based on its dns name, returns best match
+//i.e. if the record has dns name "test.sub.example.com" and route53 has two hosted zones "example.com" and "sub.example.com"
+//"sub.example.com" will be returned
+func getZoneIDForEndpoint(hostedZonesMap map[string]string, record *route53.ResourceRecordSet) string {
```

Comment:
`test.sub.example.com` will be used as NS, and upserting will result in an error.